src/biotite/structure/io/mol2/file.py

```diff
@@ -0,0 +1,990 @@
+        3: "3",
+        5: "ar",
+        0: "un",                    
+    }
+
+    supported_charge_types = [
+        "NO_CHARGES", "DEL_RE", "GASTEIGER", 
+        "GAST_HUCK", "HUCKEL", "PULLMAN", 
+        "GAUSS80_CHARGES", "AMPAC_CHARGES", 
+        "MULLIKEN_CHARGES", "DICT_CHARGES", 
+        "MMFF94_CHARGES", "USER_CHARGES"
+    ]
+        
+        
+    supported_mol_types = [
+        "SMALL", "BIOPOLYMER", "PROTEIN", "NUCLEIC_ACID", "SACCHARIDE"
+    ]    
+        
+    sybyl_status_bit_types = [
+        "system", "invalid_charges", "analyzed", "substituted", "altered", "ref_angle"
+    ]
+
+            
+    # field that contains all accepted two letter elements
+    # this is used for deriving if sperate atom_name and element 
+    # columns should be generated. If only two letter atom_name entries in the 
+    # mol2 file are found that are contained here, the atom_name entry will be
+    # assumed to only contain element names.        
+    elements_twoLetters = [
+        "LI", "NA", "MG", "AL",
+        "SI", "CA", "CR", "MN",
+        "FE", "CO", "CU", "CL",
+        "BR", "ZN", "SE", "MO",
+        "SN", "AR"
+    ]    
+    
+    @staticmethod        
+    def get_sybyl_atom_type(atom, bonds, atom_id):
+        """
+        This (horible horribel) function is meant to translate all occuring 
+        atoms into sybyl atom types based on their element and bonds. This
+        mostly works now but some of the special cases are not implemented.
+        Also it is not clear if there is some further special handling for
+        the two letter element atoms or if their sybyl_atom_type is simply
+        the same string with the second letter to lower case.
+        
+        Parameters
+        ----------
+            atom : Atom
+                The Atom object which is to be translated 
+                into sybyl atom_type notation
+            
+            bonds: BondList
+                The BondList object of the respective bonds in the AtomArray.
+                Necessary as the sybyl atom types depend on the hybridiziation 
+                heuristic.
+                
+            atom_id: int
+                The id of the current atom that is used in the bond list.                    
+                
+        Returns
+        -------     
+            sybyl_atom_type : str
+                The name of the atom based on hybridization and element according
+                to the sybyl atom types.       
+            
+        """
+        atom_bonds, types = bonds.get_bonds(atom_id)    
+        
+        if atom.element == "C":                          
+            if 5 in types:
+                return "C.ar"
+            else:
+                if len(atom_bonds) == 3:
+                    return "C.1"
+                elif len(atom_bonds) == 2:
+                    return "C.2"
+                elif len(atom_bonds) == 1:
+                    return "C.3"                         
```

Comment:
I think the hybridization is wrong here: For example, if the carbon has two bonds it should be sp (C.1). The case of one bond should in theory never occur.

Reply:
I corrected this but I have to say I really do not understand this. Based on the different hybridization being determined by number of binding, I should get N_bonds==2, N_bonds==3, N_bonds==4 in the part of the heuristic where "N" is being processed. However if I do that  some of the test files already have wrong atom_type entries.